### DISCUSSION TITLE
Is there any way to fetch S3 object with respect to Header Attribute value ? (x-amz-meta-)

### DISCUSSION THREAD

**Discussion by dnyaneshwar-jadhav-pki** [aws/aws-sdk-java-v2#5080] [Q&A]
I am storing my S3 object with Header attribute for example `x-amz-meta-x-title` as a , there are several with this way across multiple objects & multiple buckets.
In order to fetch S3 object with AWS Version is very much straight forward,
But here I am wanted to fetch the S3 whose input value to be search match with `x-amz-meta-x-title` value.

I am trying with existing approach which is looping through `ListObjectVersionsIterable` and then match in every iteration.
This is slight efforts consuming because I have to iterate over long list of Object meta data. and then found searched element.
If possible AWS SDK provide some API where in REST api I can request to fetch out object w.r.t. Header attribute keys (Of course that header key should be available)

**Comment by debora-ito** [comment]
Hi @dnyaneshwar-jadhav-pki 

I don't think there's an API that performs a search based on object metadata, I looked quickly through the S3 Select and Athena APIs and docs and couldn't find one. You're existing approach is what I'd suggest.
